File: frontend/src/api/jumpserver.js
import api from './index'

// ==================== 凭证管理 ====================
export const getCredentials = (params) => api.get('/jumpserver/credentials', { params })
export const getAllCredentials = () => api.get('/jumpserver/credentials/all')
export const getCredential = (id) => api.get(`/jumpserver/credentials/${id}`)
export const createCredential = (data) => api.post('/jumpserver/credentials', data)
export const updateCredential = (id, data) => api.put(`/jumpserver/credentials/${id}`, data)
export const deleteCredential = (id) => api.delete(`/jumpserver/credentials/${id}`)

// ==================== 主机-凭证关联 ====================
export const getHostCredentials = (hostId) => api.get(`/jumpserver/host-credentials/${hostId}`)
export const bindHostCredentials = (data) => api.post('/jumpserver/host-credentials/bind', data)

// ==================== 会话管理 ====================
export const getSessions = (params) => api.get('/jumpserver/sessions', { params })
export const getSessionStats = () => api.get('/jumpserver/sessions/stats')
export const getSession = (id) => api.get(`/jumpserver/sessions/${id}`)
export const getSessionCommands = (id, params) => api.get(`/jumpserver/sessions/${id}/commands`, { params })
export const deleteSession = (id) => api.delete(`/jumpserver/sessions/${id}`)
export const getRecordingUrl = (sessionId) => `/api/v1/jumpserver/recordings/${sessionId}`
export const getRecordingDownloadUrl = (sessionId) => `/api/v1/jumpserver/recordings/${sessionId}/download`

// ==================== 连接管理 ====================
export const connectHost = (data) => api.post('/jumpserver/connect', data)
export const disconnectSession = (sessionId) => api.post(`/jumpserver/disconnect/${sessionId}`)

// ==================== 权限管理 ====================
export const getPermissions = (params) => api.get('/jumpserver/permissions', { params })
export const getPermission = (id) => api.get(`/jumpserver/permissions/${id}`)
export const createPermission = (data) => api.post('/jumpserver/permissions', data)
export const updatePermission = (id, data) => api.put(`/jumpserver/permissions/${id}`, data)
export const deletePermission = (id) => api.delete(`/jumpserver/permissions/${id}`)
export const checkPermission = (hostId) => api.get('/jumpserver/permissions/check', { params: { hostId } })

// ==================== 审批管理 ====================
export const getApprovals = (params) => api.get('/jumpserver/approvals', { params })
export const getApproval = (id) => api.get(`/jumpserver/approvals/${id}`)
export const createApproval = (data) => api.post('/jumpserver/approvals', data)
export const approveApproval = (id, data) => api.put(`/jumpserver/approvals/${id}/approve`, data)
export const rejectApproval = (id, data) => api.put(`/jumpserver/approvals/${id}/reject`, data)

// ==================== 审计日志 ====================
export const getAuditLogs = (params) => api.get('/jumpserver/audit-logs', { params })

// ==================== 危险命令规则 ====================
export const getRiskRules = (params) => api.get('/jumpserver/risk-rules', { params })
export const getRiskRule = (id) => api.get(`/jumpserver/risk-rules/${id}`)
export const createRiskRule = (data) => api.post('/jumpserver/risk-rules', data)
export const updateRiskRule = (id, data) => api.put(`/jumpserver/risk-rules/${id}`, data)
export const deleteRiskRule = (id) => api.delete(`/jumpserver/risk-rules/${id}`)

// ==================== 批量执行 ====================
export const batchExec = (data) => api.post('/jumpserver/batch-exec', data)
export const getBatchTask = (taskId) => api.get(`/jumpserver/batch-exec/${taskId}`)

// ==================== 平台管理 ====================
export const getPlatforms = () => api.get('/jumpserver/platforms')

File: backend/internal/routes/jumpserver/jumpserver.go
package jumpserver

import (
	jumpserverCtrl "devops-console-backend/internal/controllers/jumpserver"
	"devops-console-backend/internal/dal/mapper"
	"devops-console-backend/internal/services/jumpserver"
	"devops-console-backend/pkg/configs"

	"github.com/gin-gonic/gin"
	"gorm.io/gorm"
)

// RegisterJumpserverRouters 注册跳板机路由
func RegisterJumpserverRouters(apiGroup *gin.RouterGroup, externalDB ...*gorm.DB) {
	var db *gorm.DB
	if len(externalDB) > 0 && externalDB[0] != nil {
		db = externalDB[0]
	} else {
		db = configs.NewDB()
	}

	// Mapper
	credentialMapper := mapper.NewJumpserverCredentialMapper(db)
	hostCredentialMapper := mapper.NewJumpserverHostCredentialMapper(db)
	sessionMapper := mapper.NewJumpserverSessionMapper(db)
	commandMapper := mapper.NewJumpserverCommandMapper(db)
	permissionMapper := mapper.NewJumpserverAssetPermissionMapper(db)
	approvalMapper := mapper.NewJumpserverApprovalMapper(db)
	auditLogMapper := mapper.NewJumpserverAuditLogMapper(db)
	riskRuleMapper := mapper.NewJumpserverRiskRuleMapper(db)
	platformMapper := mapper.NewJumpserverPlatformMapper(db)
	hostMapper := mapper.NewAssetHostMapper(db)

	// SSH Proxy
	sshProxy := jumpserver.NewSSHProxy(db, "./storage/replays")

	// Batch Executor
	batchExecutor := jumpserver.NewBatchExecutor(db)

	// Controllers
	credentialCtrl := jumpserverCtrl.NewCredentialController(credentialMapper)
	sessionCtrl := jumpserverCtrl.NewSessionController(sessionMapper, commandMapper, sshProxy)
	permissionCtrl := jumpserverCtrl.NewPermissionController(permissionMapper, hostMapper)
	approvalCtrl := jumpserverCtrl.NewApprovalController(approvalMapper, hostMapper, permissionMapper)
	auditCtrl := jumpserverCtrl.NewAuditController(auditLogMapper, riskRuleMapper)
	batchCtrl := jumpserverCtrl.NewBatchController(batchExecutor, hostMapper)
	platformCtrl := jumpserverCtrl.NewPlatformController(platformMapper)

	jsGroup := apiGroup.Group("/jumpserver")
	{
		// ===== 凭证管理 =====
		credGroup := jsGroup.Group("/credentials")
		{
			credGroup.GET("", credentialCtrl.ListCredentials)
			credGroup.GET("/all", credentialCtrl.ListAllCredentials)
			credGroup.GET("/:id", credentialCtrl.GetCredential)
			credGroup.POST("", credentialCtrl.CreateCredential)
			credGroup.PUT("/:id", credentialCtrl.UpdateCredential)
			credGroup.DELETE("/:id", credentialCtrl.DeleteCredential)
		}

		// ===== 主机-凭证关联 =====
		jsGroup.GET("/host-credentials/:hostId", func(c *gin.Context) {
			id, _ := jumpserverCtrl.ParseUint64Param(c, "hostId")
			list, err := hostCredentialMapper.GetByHostID(id)
			if err != nil {
				c.JSON(200, gin.H{"status": 500, "message": err.Error(), "data": nil})
				return
			}
			c.JSON(200, gin.H{"status": 200, "message": "success", "data": gin.H{"data": list}})
		})
		jsGroup.POST("/host-credentials/bind", func(c *gin.Context) {
			var req struct {
				HostID        uint64   `json:"hostId" binding:"required"`
				CredentialIDs []uint64 `json:"credentialIds" binding:"required"`
			}
			if err := c.ShouldBindJSON(&req); err != nil {
				c.JSON(200, gin.H{"status": 400, "message": err.Error(), "data": nil})
				return
			}
			if err := hostCredentialMapper.BindHostCredentials(req.HostID, req.CredentialIDs); err != nil {
				c.JSON(200, gin.H{"status": 500, "message": err.Error(), "data": nil})
				return
			}
			c.JSON(200, gin.H{"status": 200, "message": "success", "data": nil})
		})

		// ===== 会话管理 =====
		sessionGroup := jsGroup.Group("/sessions")
		{
			sessionGroup.GET("", sessionCtrl.ListSessions)
			sessionGroup.GET("/stats", sessionCtrl.GetSessionStats)
			sessionGroup.GET("/:id", sessionCtrl.GetSession)
			sessionGroup.GET("/:id/commands", sessionCtrl.GetSessionCommands)
			sessionGroup.DELETE("/:id", sessionCtrl.DeleteSession)
		}

		// ===== 连接管理 =====
		jsGroup.POST("/connect", sessionCtrl.Connect)
		jsGroup.POST("/disconnect/:sessionId", sessionCtrl.Disconnect)

		// ===== 录像回放 =====
		jsGroup.GET("/recordings/:sessionId", sessionCtrl.GetSessionRecording)
		jsGroup.GET("/recordings/:sessionId/download", sessionCtrl.DownloadRecording)

		// ===== 权限管理 =====
		permGroup := jsGroup.Group("/permissions")
		{
			permGroup.GET("", permissionCtrl.ListPermissions)
			permGroup.GET("/check", permissionCtrl.CheckPermission)
			permGroup.GET("/:id", permissionCtrl.GetPermission)
			permGroup.POST("", permissionCtrl.CreatePermission)
			permGroup.PUT("/:id", permissionCtrl.UpdatePermission)
			permGroup.DELETE("/:id", permissionCtrl.DeletePermission)
		}

		// ===== 审批管理 =====
		approvalGroup := jsGroup.Group("/approvals")
		{
			approvalGroup.GET("", approvalCtrl.ListApprovals)
			approvalGroup.GET("/:id", approvalCtrl.GetApproval)
			approvalGroup.POST("", approvalCtrl.CreateApproval)
			approvalGroup.PUT("/:id/approve", approvalCtrl.Approve)
			approvalGroup.PUT("/:id/reject", approvalCtrl.Reject)
		}

		// ===== 审计日志 =====
		jsGroup.GET("/audit-logs", auditCtrl.ListAuditLogs)

		// ===== 危险命令规则 =====
		riskGroup := jsGroup.Group("/risk-rules")
		{
			riskGroup.GET("", auditCtrl.ListRiskRules)
			riskGroup.GET("/:id", auditCtrl.GetRiskRule)
			riskGroup.POST("", auditCtrl.CreateRiskRule)
			riskGroup.PUT("/:id", auditCtrl.UpdateRiskRule)
			riskGroup.DELETE("/:id", auditCtrl.DeleteRiskRule)
		}

		// ===== 批量执行 =====
		jsGroup.POST("/batch-exec", batchCtrl.BatchExec)
		jsGroup.GET("/batch-exec/:taskId", batchCtrl.GetBatchTask)

		// ===== 平台管理 =====
		jsGroup.GET("/platforms", platformCtrl.ListPlatforms)
	}

	// 返回 SSHProxy 实例，供 WebSocket 路由使用
	// (通过包级变量共享)
	SetSSHProxy(sshProxy)
}

// 包级变量用于共享 SSHProxy 实例
var globalSSHProxy *jumpserver.SSHProxy

// SetSSHProxy 设置全局 SSHProxy
func SetSSHProxy(proxy *jumpserver.SSHProxy) {
	globalSSHProxy = proxy
}

// GetSSHProxy 获取全局 SSHProxy
func GetSSHProxy() *jumpserver.SSHProxy {
	return globalSSHProxy
}

// ParseUint64Param 导出解析参数函数
func ParseUint64Param(c *gin.Context, key string) (uint64, error) {
	return jumpserverCtrl.ParseUint64Param(c, key)
}